src/doc/tutorials/CCL.rst

```diff
@@ -0,0 +1,266 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Make Window 2 active (The window with the Isovolume operator pipeline )
+
+   .. figure:: images/ccl_win_2_active.png
+
+      Changing active window to 2
+
+#. Repeat the steps above to add a *DeferExpression* operator and set it up to defer the *ccl* expression.
+
+#. Use the Variables menu to change the active variable to *ccl*.
+
+   .. figure:: images/ccl_sel_ccl_expr.png
+
+      Changing active plot variable to *ccl*.
+
+#. Optionally, open the Pseudocolor attributes Window and change the color table used. These plots use the ``Spectral`` color table.
+
+#. Click *Draw*. You will now see the mesh rendered with Connected Component Labels.
+
+   .. figure:: images/ccl_of_isov.png
+
+      Connected Components of the Isovolume Result
+
+Again, we now have Pseudocolor plot of a new scalar field where each element is associated with its connected component label.
+In this case we have 19 connected components labeled using ids 0 - 18. You can lock the views between the two windows
+to compare the differences in the meshes and identify where the connected components differ.
+
+
+Using the Connected Component Summary query
+-------------------------------------------------
+
+The Connected Components Summary query computes and aggregates all component info and optionally writes this data to an okc file (an XMDF File format) 
+
+The query returns the following details of each component:
+
+.. list-table::
+   :widths: 40 30
+   :header-rows: 1
+
+   * - Component Data
+     - Fields
+   * - Component Id
+     - comp_label
+   * - Centroid
+     - comp_x, comp_y, comp_z
+   * - Number of Cells (or Zones) per component
+     - comp_num_cells
+   * - Area (if the dataset is 2D)
+     - comp_area
+   * - Volume (if the dataset is 3D or RZ)
+     - comp_volume
+   * - Variable Sum
+     - comp_sum
+   * - Weighted Variable Sum
+     - comp_weighted_sum
+   * - Spatial Bounding Box
+     - comp_bb_x_min, comp_bb_x_max, comp_bb_y_min, comp_bb_y_max, comp_bb_z_min, comp_bb_z_max
+   * - Number of MPI Tasks spanned by the component 
+     - comp_num_procs
+
+
+Next, we use the Connected Components Summary via python on one of our plot to obtain this info.
+
+
+#. Launch VisIt's Command Line Interface (CLI) (*Controls Menu->Launch CLI*)
+
+   .. figure:: images/ccl_launch_cli.png
+
+      Launch the CLI
+
+
+#. Run the following code snippets
+
+   .. code-block:: python
+
+       # Execute our connected components query and get the result 
+       Query("Connected Components Summary")
+       res = GetQueryOutputObject()
+
+   .. code-block:: python
+
+       # Show names in the results dictionary
+       print(res.keys())
+
+   .. code-block:: python
+
+       # Print the array of per component volumes
+       print(res["comp_volume"])
+
+   .. code-block:: python
+
+       # Print all of the results in a ~human friendly way
+       import json
+       print(json.dumps(res,indent=2))
+
+
```

Comment:
Did you want to show the results of the print commands (or maybe just the last one)?

Reply:
reformatted this and added output, i kept the output out for the last command since it is super verbose, but I put it in for the others